### ISSUE TITLE
Generate methods does not work if an interface is passed as the Generic type

### ISSUE THREAD

**Issue by yadviv** [bchavez/Bogus#372]
### Please describe why you are requesting a feature
I want to create a fake object which is an instance of an interface using Faker library, for e.g. `var faker = new Faker<Interface>()`
I guess this not possible at the moment, atleast I am not aware of it.
FYI: I am new to C# with an experience in the other programming language such as: Java, JavaScript/NodeJS and Python.

#### Working Code Implementation:
```
using Bogus;

namespace Tutorial
{
    internal static class HelloFaker
    {
        private static void Main(string[] args)
        {
            var myInterfaceFaker = new MyInterfaceFaker();
            var myInterface = myInterfaceFaker.Generate();
        }
    }

    public interface IMyInterface
    {
        public string FirstName { get; set; }
        public string LastName { get; set; }
    }

    public class MyInterfaceFaker
    {
        private readonly Faker<IMyInterface> _faker;

        public MyInterfaceFaker()
        {
            _faker = new Faker<IMyInterface>();
            var person = new Person();
            _faker.RuleFor(i => i.FirstName, () => person.FirstName);
            _faker.RuleFor(i => i.LastName, () => person.LastName);
        }

        public IMyInterface Generate()
        {
            return _faker.Generate();
        }
    }
}
```
#### Runtime Exception:
```
Unhandled exception. System.MissingMethodException: Cannot create an instance of an interface.
   at System.RuntimeTypeHandle.CreateInstance(RuntimeType type, Boolean publicOnly, Boolean wrapExceptions, Boolean& canBeCached, RuntimeMethodHandleInternal& ctor, Boolean& hasNoDefaultCtor)
   at System.RuntimeType.CreateInstanceDefaultCtorSlow(Boolean publicOnly, Boolean wrapExceptions, Boolean fillCache)
   at System.RuntimeType.CreateInstanceDefaultCtor(Boolean publicOnly, Boolean skipCheckThis, Boolean fillCache, Boolean wrapExceptions)
   at System.Activator.CreateInstance[T]()
   at Bogus.Faker`1.<>c.<.ctor>b__28_0(Faker faker)
   at Bogus.Faker`1.Generate(String ruleSets)
   at Tutorial.MyInterfaceFaker.Generate() in ....HelloFaker.cs:line 35
   at Tutorial.HelloFaker.Main(String[] args) in ....\HelloFaker.cs:line 11

Process finished with exit code -532,462,766.

```

_(Please be complete. Include all code necessary to run your example in LINQPad.)_
_(The more complete code examples are, the more accurate answers will be.)_
_(https://www.linqpad.net)_

### Please answer any or all of the questions below

* Is the feature something that currently cannot be done?
   * Yes (Possibly, library limitation)

* What alternatives have you considered?
  * To create the instance of the interface with fake data, I have to create a new concrete class implementing the desired interface.
  * Suggestion: In Java, dynamic proxies are considered to perform such kind of task.

* Is this feature request any issues or current problems?
   * It is more like a limitation from my point of view.

* Has the feature been requested in the past?
  * I guess, No

**If the feature request is approved, would you be willing to submit a PR?**
Yes. I would love to.

**Comment by bchavez** [comment]
You're correct; **Bogus** does not support `new Faker<IInterface>` out-of-the-box.

You'll need to use some underlying mock library to help you create _instances_ of **_interfaces_** without concrete types. You have a couple of options:

## Option A: Using underlying mock library

The following works with [**NSubsitute**](https://nsubstitute.github.io/):

```csharp
void Main()
{
   var personFaker = new Faker<IPerson>()
      .CustomInstantiator(_ =>
      {
         return Substitute.For<IPerson>();
      })
      .RuleFor(p => p.FirstName, f => f.Name.FirstName())
      .RuleFor(p => p.LastName, _ => "Chavez");
   personFaker.Generate().Dump();
}
public interface IPerson
{
   string FirstName { get; set; }
   string LastName { get; set; }
}
```
![image](https://user-images.githubusercontent.com/478118/118197617-28a98800-b404-11eb-8d44-20084efa112e.png)

## Option B: Nick Dodd's AutoBogus library
Using Nick Dodd's [AutoBogus](https://github.com/nickdodd79/AutoBogus) library; specifically, referencing [AutoBogus.FakeItEasy](https://www.nuget.org/packages/AutoBogus.FakeItEasy); the following works without concrete types:
```csharp
void Main()
{
   var personFaker = new AutoFaker<IPerson>()
      .Configure( config => config.WithBinder<FakeItEasyBinder>())
      .RuleFor( p => p.FirstName, f => f.Name.FirstName() )
      .RuleFor( p => p.LastName, _ => "Chavez");
      
   personFaker.Generate().Dump();
}
public interface IPerson{
   string FirstName {get;set;}
   string LastName {get;set;}
}
```
![image](https://user-images.githubusercontent.com/478118/118197149-34e11580-b403-11eb-9f71-742c2487c673.png)

In summary, you need ***something*** backing your interface; otherwise, `Faker<T>` won't work. **Bogus** is pretty basic like that.

**Comment by yadviv** [comment]
I used [NSubsitute](https://nsubstitute.github.io/), and it worked for me. Thank you @bchavez. 
I added `.CustomInstantiator(_ => Substitute.For<IMyInterface>())` into my implementation example code, as shown below.
```
     ....
     _faker = new Faker<IMyInterface>().CustomInstantiator(_ => Substitute.For<IMyInterface>());
     ....
```

**Comment by guercheLE** [comment]
> You're correct; **Bogus** does not support `new Faker<IInterface>` out-of-the-box.
> 
> You'll need to use some underlying mock library to help you create _instances_ of **_interfaces_** without concrete types. You have a couple of options:
> 
> ## Option A: Using underlying mock library
> The following works with [**NSubsitute**](https://nsubstitute.github.io/):
> 
> ```cs
> void Main()
> {
>    var personFaker = new Faker<IPerson>()
>       .CustomInstantiator(_ =>
>       {
>          return Substitute.For<IPerson>();
>       })
>       .RuleFor(p => p.FirstName, f => f.Name.FirstName())
>       .RuleFor(p => p.LastName, _ => "Chavez");
>    personFaker.Generate().Dump();
> }
> public interface IPerson
> {
>    string FirstName { get; set; }
>    string LastName { get; set; }
> }
> ```
> 
> ![image](https://user-images.githubusercontent.com/478118/118197617-28a98800-b404-11eb-8d44-20084efa112e.png)
> 
> ## Option B: Nick Dodd's AutoBogus library
> Using Nick Dodd's [AutoBogus](https://github.com/nickdodd79/AutoBogus) library; specifically, referencing [AutoBogus.FakeItEasy](https://www.nuget.org/packages/AutoBogus.FakeItEasy); the following works without concrete types:
> 
> ```cs
> void Main()
> {
>    var personFaker = new AutoFaker<IPerson>()
>       .Configure( config => config.WithBinder<FakeItEasyBinder>())
>       .RuleFor( p => p.FirstName, f => f.Name.FirstName() )
>       .RuleFor( p => p.LastName, _ => "Chavez");
>       
>    personFaker.Generate().Dump();
> }
> public interface IPerson{
>    string FirstName {get;set;}
>    string LastName {get;set;}
> }
> ```
> 
> ![image](https://user-images.githubusercontent.com/478118/118197149-34e11580-b403-11eb-9f71-742c2487c673.png)
> 
> In summary, you need _**something**_ backing your interface; otherwise, `Faker<T>` won't work. **Bogus** is pretty basic like that.

Tried the sample which uses .Configure( config => config.WithBinder<FakeItEasyBinder>()), but without any .RuleFor. It created, but did not populated properties.

Any clues on how to not have to define many .RuleFor on interfaces which have many properties?
